otlp/src/otlp/logs/logger.cpp

```diff
@@ -225,17 +251,26 @@ void Logger::SendingLoop(Queue::Consumer& consumer, LogClient& log_client, Trace
                     [&scope_logs](const opentelemetry::proto::logs::v1::LogRecord& action) {
                         auto log_records = scope_logs->add_log_records();
                         *log_records = action;
-                    }},
+                    }
+                },
                 action
             );
         } while (consumer.Pop(action, deadline));
 
+        std::vector<engine::TaskWithResult<void>> tasks(2);
+
         if (config_.logs_sink == SinkType::kBoth || config_.logs_sink == SinkType::kOtlp) {
-            DoLog(log_request, log_client);
+            tasks.emplace_back(engine::CriticalAsyncNoSpan([this, &log_request, &log_client]() {
+                DoLog(log_request, log_client);
+            }));
         }
         if (config_.tracing_sink == SinkType::kBoth || config_.tracing_sink == SinkType::kOtlp) {
-            DoTrace(trace_request, trace_client);
+            tasks.emplace_back(engine::CriticalAsyncNoSpan([this, &trace_request, &trace_client]() {
+                DoTrace(trace_request, trace_client);
+            }));
         }
+
+        engine::GetAll(tasks);
```

Comment:
why do we need these async calls?

Reply:
Because retries are performed there, so as not to block the execution of one log sending while traces are being sent